stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -376,6 +376,27 @@ function SparseMatrixCSC{Tv,Ti}(M::AbstractMatrix) where {Tv,Ti}
     eltypeTvV = convert(Vector{Tv}, V)
     return sparse_IJ_sorted!(eltypeTiI, eltypeTiJ, eltypeTvV, size(M)...)
 end
+function SparseMatrixCSC{Tv,Ti}(M::Matrix) where {Tv,Ti}
+    nz = count(!equalto(0), M)
```

Comment:
`tv` might not be concrete, and the matrix may contain `missing` or (more realistically) be have SVector entries. 

Maybe `nz = count(x->true===iszero(x), M)`? That way we only lose type information of zeros.

Not sure about performance implications (sorry for double-posting, failed at github-ui).

Reply:
No, this is a pure performance improvement over the current function which uses `findnz` which does `nnzA = count(t -> t != 0, A)` and `if Aij != 0`. Changing the sparse matrix library to work more complicated elements is out of scope for this PR.

Reply:
Note that `equalto` uses `isequal`, so this will change the behavior for `-0.0`.